src/core/MOM_stoch_eos.F90

```diff
@@ -0,0 +1,212 @@
+  type(diag_ctrl),       target, intent(inout) :: diag       !< to control diagnostics
+  integer :: i,j
+  type(vardesc)      :: vd
+  seed=0
+  ! contants
+  !pi=2*acos(0.0)
+  call get_param(param_file, "MOM_stoch_eos", "STOCH_EOS", stoch_eos_CS%use_stoch_eos, &
+                 "If true, stochastic perturbations are applied "//&
+                 "to the EOS in the PGF.", default=.false.)
+  call get_param(param_file, "MOM_stoch_eos", "STANLEY_COEFF", stoch_eos_CS%stanley_coeff, &
+                 "Coefficient correlating the temperature gradient "//&
+                 "and SGS T variance.", default=-1.0)
+  call get_param(param_file, "MOM_stoch_eos", "STANLEY_A", stoch_eos_CS%stanley_a, &
+                 "Coefficient a which scales chi in stochastic perturbation of the "//&
+                 "SGS T variance.", default=1.0)
+  call get_param(param_file, "MOM_stoch_eos", "KD_SMOOTH", stoch_eos_CS%kappa_smooth, &
+                 "A diapycnal diffusivity that is used to interpolate "//&
+                 "more sensible values of T & S into thin layers.", &
+                 units="m2 s-1", default=1.0e-6)
+
+  !don't run anything if STANLEY_COEFF < 0
+  if (stoch_eos_CS%stanley_coeff >= 0.0) then
+
+    ALLOC_(stoch_eos_CS%pattern(G%isd:G%ied,G%jsd:G%jed)) ; stoch_eos_CS%pattern(:,:) = 0.0
+    vd = var_desc("stoch_eos_pattern","nondim","Random pattern for stoch EOS",'h','1')
+    call register_restart_field(stoch_eos_CS%pattern, vd, .false., restart_CS)
+    ALLOC_(stoch_eos_CS%phi(G%isd:G%ied,G%jsd:G%jed)) ; stoch_eos_CS%phi(:,:) = 0.0
+    ALLOC_(l2_inv(G%isd:G%ied,G%jsd:G%jed))
+    ALLOC_(rgauss(G%isd:G%ied,G%jsd:G%jed))
+    call get_param(param_file, "MOM_stoch_eos", "SEED_STOCH_EOS", seed, &
+                 "Specfied seed for random number sequence ", default=0)
+    call random_2d_constructor(rn_CS, G%HI, Time, seed)
+    call random_2d_norm(rn_CS, G%HI, rgauss)
+    ! fill array with approximation of grid area needed for decorrelation
+    ! time-scale calculation
+    do j=G%jsc,G%jec
+       do i=G%isc,G%iec
+          l2_inv(i,j)=1.0/(G%dxT(i,j)**2+G%dyT(i,j)**2)
+       enddo
+    enddo
+    if (is_new_run(restart_CS)) then
+       do j=G%jsc,G%jec
+          do i=G%isc,G%iec
+             stoch_eos_CS%pattern(i,j)=amplitude*rgauss(i,j)
+          enddo
+       enddo
+    endif
+
+    !register diagnostics
+    stoch_eos_CS%id_tvar_sgs = register_diag_field('ocean_model', 'tvar_sgs', diag%axesTL, Time, &
+      'Parameterized SGS Temperature Variance ', 'None')
+    if (stoch_eos_CS%use_stoch_eos) then
+      stoch_eos_CS%id_stoch_eos = register_diag_field('ocean_model', 'stoch_eos', diag%axesT1, Time, &
+        'random pattern for EOS', 'None')
+      stoch_eos_CS%id_stoch_phi = register_diag_field('ocean_model', 'stoch_phi', diag%axesT1, Time, &
+        'phi for EOS', 'None')
+    endif
+  endif
+
+  end subroutine MOM_stoch_eos_init
+
+  subroutine MOM_stoch_eos_run(G,u,v,delt,Time,stoch_eos_CS,diag)
```

Comment:
Please indent the contents of a subroutine by two spaces relative to the declaration of the subroutine, to promote the readability of the code.

  In almost all of the rest of the MOM6 code, we start the subroutine or function declaration in the first column, as the `module` and `end module` statements should be at the start and end of the file, so indenting everything inside these does nothing to promote readability.  In all other cases, code blocks should use a consistent two-point indent scheme.  In fact, we have plans to automatically enforce this by detecting code lines that start in columns 1, 3, 5